### DISCUSSION TITLE
Props are unreasonable with HomeHero component which exposed to MDX

### DISCUSSION THREAD

**Discussion by MickeyMiao7** [web-infra-dev/rspress#1388] [General]
官网上，HomeHero 组件的参数是：
![image](https://github.com/user-attachments/assets/0ecdbb93-de8e-49f0-8b4e-16bbcb55c765)

但实际在 MDX 中引入该组件并使用，需要将参数包裹在 `frontmatter={{ hero: Hero }}` 中，使用起来很不方便。

```
import { HomeHero } from '@theme';

<HomeHero
  frontmatter={{
    hero: {
      name: 'Sample'
    }
  }}
/>
```

**Comment by Timeless0911** [comment]
Good advice, we will improve this in the future. Since it may introduce a breaking change, we plan to ship it in Rspress v2.

**Reply by MickeyMiao7** [reply]
is it possible that adding the interface `hero` in Rspress v1. In the same time, we still keep the original interface `frontmatter` and mark it as ``will be deprecated in v2``

**Reply by Timeless0911** [reply]
Rspress v2 is already in the plan, there is no need to make such changes now, since the final API structure is not determined.

**Reply by MickeyMiao7** [reply]
i see. thanks for sharing the info
